### PR TITLE
Update YASM to fix the command error

### LINKED ISSUE #12237: [gmp:x64-windows] build failure
**Host Environment**
- OS: [Microsoft Windows [Version 10.0.19041.331]]
- Compiler: Visual Studio 2019 Version: 16.6.3  VS Tools Version: 14.26.28801 

**To Reproduce**
`SYSTEM@ANDROMEDA 07/02/2020 21:20:49 D:\Git\vs2019.prod`
`->vcpkg install gmp[core]:x64-windows nettle[core]:x64-windows`

Additionally, attach any relevant sections from the log files above.

**Failure logs** 
- Cut and past the appropriate build messages from the console output.
`Computing installation plan...`
`The following packages will be built and installed:`
`    gmp[core]:x64-windows`
`    nettle[core]:x64-windows`
`Starting package 1/2: gmp:x64-windows`
`Building package gmp[core]:x64-windows...`
`-- Using cached D:/Git/vcpkg.downloads/ShiftMediaProject-gmp-e140dfc8668e96d7e56cbd46467945adcc6b3cc4.tar.gz`
`-- Extracting source D:/Git/vcpkg.downloads/ShiftMediaProject-gmp-e140dfc8668e96d7e56cbd46467945adcc6b3cc4.tar.gz`
`-- Applying patch vs.build.patch`
`-- Applying patch runtime.patch`
`-- Applying patch prefix.patch`
`-- Using source at D:/Git/vs2019.prod/buildtrees/gmp/src/adcc6b3cc4-2b6258d695`
`-- Building SMP/libgmp.sln for Release`
`-- Building SMP/libgmp.sln for Debug`
`-- Installing: D:/Git/vs2019.prod/packages/gmp_x64-windows/share/gmp/copyright`
`CMake Error at ports/gmp/portfile.cmake:61 (file):`
`  file RENAME failed to rename`

`    D:/Git/vs2019.prod/buildtrees/gmp/x64-windows-rel/adcc6b3cc4-2b6258d695/msvc/include`

`  to`

`    D:/Git/vs2019.prod/packages/gmp_x64-windows/include`

`  because: File exists`

`Call Stack (most recent call first):`
`  scripts/ports.cmake:76 (include)`

`Error: Building package gmp:x64-windows failed with: BUILD_FAILED`
`Please ensure you're using the latest portfiles with `.\vcpkg update`, then`
`submit an issue at https://github.com/Microsoft/vcpkg/issues including:`
`  Package: gmp:x64-windows`
`  Vcpkg version: 2020.02.04-nohash`

- Please attach any additional failure logs mentioned in the console output.
See #11238

**Additional context**
It appears gmp is trying to install its header (include) files in `<vcpkg-root>\installed\x64-windows\include\include` -- which fails because I also have the LLGL port installed and it also has installed its header (include) files into `<vcpkg-root>\installed\x64-windows\include\include\LLGL` (a separate issue I will submit under a separate issue. 

Both are incorrect, I believe the gmp port files should just be in the `<vcpkg-root>\install\x64-windows\include` and LLGL port should be in `<vcpkg-port>\installed\x64-windows\include\LLGL`.

In other words there is one too many `include` in the file destination paths.

### LINKED ISSUE #12240: [nettle:x64-windows] build failure
**Host Environment**
- OS: [Microsoft Windows [Version 10.0.19041.331]]
- Compiler: 
   - Visual Studio 2017 Version: 15.9.24  VS Tools Version: 14.16.27023 
   - Visual Studio 2019 Version: 16.6.3 VS Tools Version: 14.26.28801

**To Reproduce**
`vcpkg install nettle[core]:x64-windows`

**Failure logs** 
- Cut and past the appropriate build messages from the console output.
`Computing installation plan...`
`The following packages will be built and installed:`
`    nettle[core]:x64-windows`
`Starting package 1/1: nettle:x64-windows`
`Building package nettle[core]:x64-windows...`
`-- Using cached D:/Git/vcpkg.downloads/ShiftMediaProject-nettle-1d0a6e64e01458fdf37eaf5d90975deb52c3da41.tar.gz`
`-- Using source at D:/Git/vs2019.prod/buildtrees/nettle/src/eb52c3da41-e03f56976f`
`-- Building SMP/libnettle.sln for Release`
`CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):`
`    Command failed: msbuild D:/Git/vs2019.prod/buildtrees/nettle/x64-windows-rel/eb52c3da41-e03f56976f/SMP/libnettle.sln /p:Configuration=ReleaseDLL /p:UseEnv=True /t:Rebuild /p:Platform=x64 /p:PlatformToolset=v142 /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /p:WindowsTargetPlatformVersion=10.0.19041.0 /p:VcpkgTriplet=x64-windows /p:VcpkgCurrentInstalledDir=D:/Git/vs2019.prod/installed/x64-windows /p:VcpkgManifestInstall=false /m /p:ForceImportBeforeCppTargets=D:/Git/vs2019.prod/scripts/buildsystems/msbuild/vcpkg.targets /p:VcpkgApplocalDeps=false`
`    Working Directory: D:/Git/vs2019.prod/buildtrees/nettle/x64-windows-rel/eb52c3da41-e03f56976f`
`    Error code: 1`
`    See logs for more information:`
`      D:\Git\vs2019.prod\buildtrees\nettle\build-x64-windows-rel-out.log`

`Call Stack (most recent call first):`
`  scripts/cmake/vcpkg_install_msbuild.cmake:162 (vcpkg_execute_required_process)`
`  ports/nettle/portfile.cmake:66 (vcpkg_install_msbuild)`
`  scripts/ports.cmake:76 (include)`

`Error: Building package nettle:x64-windows failed with: BUILD_FAILED`
`Please ensure you're using the latest portfiles with `.\vcpkg update`, then`
`submit an issue at https://github.com/Microsoft/vcpkg/issues including:`
`  Package: nettle:x64-windows`
`  Vcpkg version: 2020.02.04-nohash`

`Additionally, attach any relevant sections from the log files above.`
- Please attach any additional failure logs mentioned in the console output.
[vs2017-build-x64-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/4867491/vs2017-build-x64-windows-rel-out.log)
[vs2019-build-x64-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/4867492/vs2019-build-x64-windows-rel-out.log)

**Additional context**
This may/may not have relevance from issue `[<gmp>] build failure` referenced #12237 as I have a work around patch to place the `gmp.h` and `gmpxx.h` in the `<vcpkg-port>\installed\x64-windows\include` folder instead of the `<vcpkg-port>\installed\x64-windows\include\include` folder.

### PR DESCRIPTION
**Pull request by JackBoosY**
Using the new YASM binary seems to solve the path problem in the command line, so update it.

Fixes #12240 #12237.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@Voskrese @Neumann-A @timautry @Sibras Could you please test this PR?

Thanks.

**Comment by JackBoosY** [comment]
I can't repro these regressions on my machine.

**Comment by Voskrese** [comment]
`portfile.cmake`


```
        OPTIONS
+          "/p:YASMPATH=${YASM_DIR}/"
```

**Comment by JackBoosY** [comment]
My log is:
`"F:/12324/vcpkg/downloads/tools/yasm/1.3.0.6/"yasm.exe -Xvc -f win64 -o ...`

The pipeline log is:
`"D:/downloads/tools/yasm/1.3.0.6/"yasm.exe -Xvc -f win64 -o ...`

But I can build successful on my machine. 
Why?

**Comment by Voskrese** [comment]
cash azure subdir yasm version x84?

subdir x64 and x86

**Comment by JackBoosY** [comment]
@Voskrese No, they are in _downloads/tools/yasm/1.3.0.6_.

**Comment by Voskrese** [comment]
x64 -- set(SUBDIR "1.3.0_x64/")

x86 -- set set(SUBDIR "1.3.0.6_x86/")

?

**Comment by JackBoosY** [comment]
@Voskrese and I found the following code doesn't work:
```
    get_filename_component(YASM_DIR "${PATHS}" DIRECTORY)
    vcpkg_add_to_path(PREPEND "${YASM_DIR}")
    set(YASMPATH "${YASM_DIR}")
```

**Comment by Voskrese** [comment]
```
    get_filename_component(YASM_DIR "${PATHS}" DIRECTORY)
    vcpkg_add_to_path(PREPEND "${YASM_DIR}")
    set(YASMPATH "${YASM_DIR}")
```

I experimented with this for myself, plus the definition of the version
Yes, it should be deleted

**Comment by JackBoosY** [comment]
@BillyONeal Could you take a look?

Thanks.

**Comment by Voskrese** [comment]
test, delete 

`             /p:UseEnv=True`

File: ports/gmp/portfile.cmake [hunk 1]
```diff
@@ -55,7 +55,9 @@ if(VCPKG_TARGET_IS_WINDOWS)
         RELEASE_CONFIGURATION ${CONFIGURATION_RELEASE}
         DEBUG_CONFIGURATION ${CONFIGURATION_DEBUG}
         SKIP_CLEAN
-        OPTIONS /p:UseEnv=True
+        OPTIONS
+            /p:UseEnv=True
```
Comment (JackBoosY) [inline ports/gmp/portfile.cmake:59]:
```suggestion
```

**Comment by Voskrese** [comment]
this parameter should not be present at all  `/p:UseEnv=True`

vcpkg too isolated, where the bug crept in
ms-mpi with many patches barely collected

**Comment by JackBoosY** [comment]
```
D:\installed\x64-windows\share\vs-yasm\yasm.targets(50,5): error MSB3721: The command ""D:/downloads/tools/yasm/1.3.0.6/x64/"yasm.exe -Xvc -f win64 -o "D:\buildtrees\gmp\x64-windows-rel\adcc6b3cc4-2b6258d695.clean\SMP\obj\ReleaseDLL\x64\libgmp\addaddmul_1msb0.obj" -rraw -pgas  "D:\buildtrees\gmp\x64-windows-rel\adcc6b3cc4-2b6258d695.clean\SMP\mpn\x86_64\addaddmul_1msb0.s"" exited with code -1073741515. 
```

I think this is a bug of yasm. I'll report it later.

**Comment by JackBoosY** [comment]
I have to wait for the official to fix this bug.

**Comment by Neumann-A** [comment]
Maybe we should just build a newer version of yasm ourself and use that?

**Comment by JackBoosY** [comment]
@Neumann-A Therefore, we have to add yasm as a port.

**Comment by Neumann-A** [comment]
@JackBoosY: Therefore, we cannot do it because we are not allowed to do so? Maybe the MS team should discuss again if vcpkg should support tools as ports or have a separate git for binary tools. I mean llvm/qmake etc. are already built by vcpkg, so why not the complete rest?

**Comment by JackBoosY** [comment]
@Neumann-A We need to discuss it.

**Comment by Voskrese** [comment]
> it may be easier to switch to the MSYS version, because it is only a tool for performing the task

yasm perl and others

pacman -S mingw-w64-x86_64-yasm

pacman -S mingw-w64-x86_64-perl

**Comment by JackBoosY** [comment]
Temporary close this PR, will continue this work in the future.

**Comment by JackBoosY** [comment]
This issue will be fixed in #14003 instead.

**Pull request closed without merging**